crates/re_data_ui/src/component.rs

```diff
@@ -85,8 +93,20 @@ impl DataUi for EntityComponentWithInstances {
                 .body(|mut body| {
                     re_ui::ReUi::setup_table_body(&mut body);
                     let row_height = re_ui::ReUi::table_line_height();
-                    body.rows(row_height, num_instances, |index, mut row| {
-                        if let Some(instance_key) = instance_keys.get(index) {
+                    body.rows(row_height, max_row, |index, mut row| {
+                        if index == max_row - 1 && num_instances > max_row {
+                            // last row, suggest that there is more.
+                            row.col(|ui| {
+                                ui.label(format!(
```

Comment:
I think it would make more sense to put this label _after_ the table body. Right now the label expands the width of the first column, but this label shouldn't need to.

Also it leads to convoluted logic where you need to use `4` to mean "at most 3 elements", which threw me off quite a bit when reading the code.

Reply:
Moving that out of the table makes total.

TBH, in the strict "max_row" sense that encompasses the "...x more" line (which I aimed for), the behaviour is intrinsically convoluted. For eg. max_row = 5, if you have 5 items you display all five, but if you have 6 you only display 4 and "...2 more".

I opted to stick to that (somewhat uptight) max_row definition but made the code much more clear.